### PR TITLE
Enhancements to page position UI

### LINKED ISSUE #3609: Page status dialog: pre-select position 1 for only public page
More detailed description: https://kirby.nolt.io/290

### PR DESCRIPTION
**Pull request by distantnative**
### Enhancements
- Page position field is hidden from dialogs when only one option available (except when in `debug` mode)
- Sort page is disabled in page dropdown when only one option would be available

## Breaking changes

None

## Related issues/ideas

- Fixes #3609

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [x] Add changes to release notes draft in Notion

### CODE REVIEW

**Comment by lukasbestle** [comment]
To be honest I'm not sure about this. If we hide the field completely in these cases, users could be confused as they expect that field to be normally there.

Possible alternative: Make the field disabled instead.

**Comment by distantnative** [comment]
We have the same behavior with the template select on page creation (hiding field if only one option is available)

**Comment by lukasbestle** [comment]
Yes, we do. I was thinking about this when I brainstormed about https://github.com/getkirby/kirby/issues/3644. But I came to the conclusion that it makes sense there as the template is just an "implementation detail" that isn't as relevant for the user. For example when you create a new blog post as the child of a blog page, you don't think about which template you need to choose. It's pretty obvious. So it makes some amount of sense to hide the selector in this case.

Now that I think about the same question here, I feel like it's more or less the same. It definitely makes sense to disable that menu option to change the position. Regarding the field in the status change dialog I also tend to agree with you now.

**Comment by lukasbestle** [comment]
But I think we should think about the different display in `debug` mode again. It can cause confusion like "why does it look different in production": https://github.com/getkirby/kirby/issues/3663

**Comment by bastianallgeier** [comment]
I agree with @lukasbestle that we don't really need the debug toggle in this case. For the template, you might want to actually "debug" which template is used. For the position it's not relevant.

**Comment by distantnative** [comment]
> I agree with @lukasbestle that we don't really need the debug toggle in this case. 

Maybe I understood Lukas wrong, but I think his argument was more that our UI should look different for the `debug` mode for both use cases/dialogs. Not that one should be shown, the other always hidden.

**Comment by bastianallgeier** [comment]
I would love to see a more prominent debug mode visual in the panel somewhere. I think this could also help to avoid unwanted debug mode in production. But in this case I would simply not make a difference between the hidden field in debug mode or prod mode.

File: src/Panel/Page.php [hunk 1]
```diff
@@ -121,11 +121,13 @@ public function dropdown(array $options = []): array
         ];
 
         if ($view === 'list') {
+            $siblings = $page->parentModel()->children()->listed()->not($page);
```
Comment (bastianallgeier) [inline src/Panel/Page.php]: Can't we use `$page->siblings()` here?

File: src/Panel/Field.php [hunk 2]
```diff
@@ -120,11 +120,20 @@ public static function pagePosition(Page $page, array $props = []): array
             'text'  => $index
         ];
 
+        // if only one available option,
+        // hide field when not in debug mode
+        if (count($options) < 2 && option('debug') !== true) {
```
Comment (bastianallgeier) [inline src/Panel/Field.php]: I'd simply remove the second check for the debug option and merge this thing.

File: src/Panel/Page.php (hunk 1 above)
Comment (bastianallgeier) [reply src/Panel/Page.php]: Forget about it. It's needed this way.

File: src/Panel/Field.php [hunk 3]
```diff
@@ -120,11 +120,20 @@ public static function pagePosition(Page $page, array $props = []): array
...
+            return array_merge([
+                'type'   => 'hidden',
+                'value'  => $options[0],
```
Comment (bastianallgeier) [inline src/Panel/Field.php]: I think this is wrong. It should probably be $options[0]['value']

**Comment by bastianallgeier** [comment]
I made a few changes. Have a look at them and let me know what you think about it. We might want to add tests, but otherwise I'd be happy with this PR

File: src/Panel/Page.php [hunk 4]
```diff
@@ -120,16 +120,14 @@ public function dropdown(array $options = []): array
             'disabled' => $this->isDisabledDropdownOption('changeStatus', $options, $permissions)
         ];
 
-        if ($view === 'list') {
```
Comment (distantnative) [inline src/Panel/Page.php:123]: You want to show this option now as well on the page itself, not just in list view?

Comment (bastianallgeier) [reply src/Panel/Page.php:123]: Yesterday when I tested it, I tried it in list view and then switched to the page view and wondered why the option isn't there at all. It felt like a bug. In this case the status dialog was disabled via permissions but sorting was still possible. So there was no way to change the position of the page via the page view at all.

Comment (distantnative) [reply src/Panel/Page.php:123]: 👍

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**